pkg/scheduler/framework/cycle_state.go

```diff
@@ -48,6 +50,8 @@ type CycleState struct {
 	storage sync.Map
 	// if recordPluginMetrics is true, PluginExecutionDuration will be recorded for this cycle.
 	recordPluginMetrics bool
+	// skipFilterPlugin is plugins that are skipped in the Filter extension point.
+	skipFilterPlugins sets.String
```

Comment:
I would just export this field instead of adding getters and setters. Or you could have functions like `SkipFilterPlugin` to add elements and `ShouldSkipFilterPlugin` to check if the plugin is in the set.

Are there scenarios in the codebase where c==nil?

Reply:
👍 
I just followed how we handle `recordPluginMetrics`, and no big reason to do this.

Reply:
> Are there scenarios in the codebase where c==nil?

I checked and seems No.